visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts

```diff
@@ -96,51 +112,78 @@ export class CodeMapLabelService implements CameraChangeSubscriber {
 		context.font = `${fontsize}px Helvetica Neue`
 
 		const margin = 20
+		const multiLineContext = message.split("\n")
 
 		// setting canvas width/height before ctx draw, else canvas is empty
 		canvas.width = context.measureText(message).width + margin
-		canvas.height = fontsize + margin
+		canvas.height = margin + fontsize * multiLineContext.length
 
 		// bg
-		context.fillStyle = "rgba(255,255,255,1)"
-		context.strokeStyle = ColorConverter.convertHexToRgba(this.storeService.getState().appSettings.mapColors.angularGreen)
+		context.font = `${fontsize}px Helvetica Neue`
+		context.fillStyle = "rgba(224,224,224,0.85)"
 		context.lineJoin = "round"
 		context.lineCap = "round"
 		context.lineWidth = 5
-		context.fillRect(0, 0, canvas.width, canvas.height)
-		context.strokeRect(0, 0, canvas.width, canvas.height)
+
+		this.drawRectangleWithRoundedCorners(context, 0, 0, canvas.width, canvas.height, this.LABEL_CORNER_RADIUS)
 
 		// after setting the canvas width/height we have to re-set font to apply!?! looks like ctx reset
-		context.font = `${fontsize}px Helvetica Neue`
 		context.fillStyle = "rgba(0,0,0,1)"
 		context.textAlign = "center"
 		context.textBaseline = "middle"
-		context.fillText(message, canvas.width / 2, canvas.height / 2)
+
+		for (const [i, element] of multiLineContext.entries()) {
+			context.fillText(element, canvas.width / 2, (canvas.height * (i + 1)) / (multiLineContext.length + 1))
+		}
 
 		const texture = new Texture(canvas)
 		texture.minFilter = LinearFilter // NearestFilter;
 		texture.needsUpdate = true
 
 		const spriteMaterial = new SpriteMaterial({ map: texture })
 		const sprite = new Sprite(spriteMaterial)
-		this.setLabelSize(sprite, canvas.width)
+		this.lineCount = multiLineContext.length
+		this.setLabelSize(sprite, canvas.width, null)
 
 		return {
 			sprite,
 			heightValue: canvas.height,
-			line: null
+			line: null,
+			lineCount: multiLineContext.length
 		}
 	}
 
-	private setLabelSize(sprite: Sprite, labelWidth: number = sprite.material.map.image.width) {
+	private drawRectangleWithRoundedCorners(context, x, y, width, height, radius) {
+		if (width < 2 * radius) radius = width / 2
```

Comment:
A comment would be nice here, also.

Reply:
I am not sure what would be commented here, its just a polygon
Each arcTo gets 4 parameters(2 points) to draw a line between and a radius for the curving
The if condition assures that the polygon can not be build not to meet (although it can not happen in our case due to radius being set and width being much higher, maybe I can delete that)

So ycould you provide an example of what I should explain? :)